### PR TITLE
Add pyribs visualization

### PR DESCRIPTION
**Pull request by btjanaka**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- [x] I agree to the contributor agreements.

> [!TIP]
> Please follow the [Quick TODO list](https://github.com/optuna/optunahub-registry/tree/main?tab=readme-ov-file#quick-todo-list-towards-contribution) to smoothly merge your PR.

## Motivation

This PR creates a plugin that provides wrappers for pyribs visualization tools. Currently, the plugin only supports the [grid_archive_heatmap](https://docs.pyribs.org/en/stable/api/ribs.visualize.grid_archive_heatmap.html) function from pyribs, which is sufficient for plotting the results from CmaMaeSampler.

## Description of the changes

* Created a new visualization plugin under `visualization/plot_pyribs`
* Added a new function to the plugin called `plot_grid_archive_heatmap`, which is a wrapper around `ribs.visualize.grid_archive_heatmap` from pyribs.

## TODO List towards PR Merge

Please remove this section if this PR is not an addition of a new package.
Otherwise, please check the following TODO list:

- [x] Copy `./template/` to create your package
- [x] Replace `<COPYRIGHT HOLDER>` in `LICENSE` of your package with your name
- [x] Fill out `README.md` in your package
- [x] Add import statements of your function or class names to be used in `__init__.py`
- [x] (Optional) Add `from __future__ import annotations` at the head of any Python files that include typing to support older Python versions
- [x] Apply the formatter based on the tips in [`README.md`](https://github.com/optuna/optunahub-registry/tree/main)
- [x] Check whether your module works as intended based on the tips in [`README.md`](https://github.com/optuna/optunahub-registry/tree/main)

### CODE REVIEW

**Comment by btjanaka** [comment]
Hi @nabenabe0928, I have added the visualization tools for plotting results from the CMA-MAE sampler. Would you take a look and let me know your thoughts? Thank you!

**Comment by nabenabe0928** [comment]
@btjanaka 

Hey, thank you for the PR!
I will take a look at your PR asap and merge this feature!

> Do you think I should put this visualization example in the CMA-MAE sampler documentation too?

It's up to you, but I think it would increase the in-flow of users to this visualization module if we have the example!

File: package/samplers/cmamae/README.md [hunk 1]
```diff
@@ -27,6 +27,9 @@ with improvement ranking, all wrapped up in a
 However, it is possible to implement many variations of CMA-MAE and other
 quality diversity algorithms using pyribs.
 
+For visualizing the results of the `CmaMaeSampler`, note that we use the
+`plot_grid_archive_heatmap` function from the `plot_pyribs` plugin.
```
Comment (btjanaka) [inline package/samplers/cmamae/README.md:31]: @nabenabe0928 What is the correct way to link between plugins in OptunaHub? i.e., How can I provide a link to the `plot_pyribs` plugin here? Similarly, in the `plot_pyribs` plugin, I want to provide a link to the `CmaMaeSampler`.

**Comment by nabenabe0928** [comment]
@btjanaka Hey sorry for the late response!
I was away from the development for a week and came back today!

File: package/samplers/cmamae/README.md (hunk 1 above)
Comment (nabenabe0928) [reply package/samplers/cmamae/README.md:31]: I will quickly merge this PR, so you can surely open another PR later, but basically, the URL for each package is published like [this](https://hub.optuna.org/samplers/cmamae/), so you could link these URLs to your package!

File: package/visualization/plot_pyribs/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+title: Pyribs Visualization Wrappers
+description: This visualizaton module provides wrappers around the visualization functions from pyribs, which is useful for plotting results from CmaMaeSampler.
+tags: [visualization, quality diversity, pyribs]
+optuna_versions: [4.0.0]
+license: MIT License
+---
+
+## Class or Function Names
+
+- plot_grid_archive_heatmap
```
Comment (nabenabe0928) [inline package/visualization/plot_pyribs/README.md:12]:
```suggestion
- `plot_grid_archive_heatmap()`

- `HEBOSampler(study: optuna.Study, ax: plt.Axes, **kwargs)`
  - `study`: Optuna study with a sampler that uses pyribs. This function will plot the result archive from the sampler's scheduler.
  - ax: Axes on which to plot the heatmap. If None, we retrieve the current axes.
  - kwargs: All remaining kwargs will be passed to [`grid_archive_heatmap`](https://docs.pyribs.org/en/stable/api/ribs.visualize.grid_archive_heatmap.html).

```

**Review by nabenabe0928 — COMMENTED** [review]
After I verify the behavior of your code, I will merge this PR!
Please link the URLs to the corresponding locations in another PR:)

**Review by nabenabe0928 — APPROVED** [review]
I confirmed that the example works!
I will merge this PR:)

File: package/visualization/plot_pyribs/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+---
+
+## Class or Function Names
+
+- `plot_grid_archive_heatmap()`
+
+- `HEBOSampler(study: optuna.Study, ax: plt.Axes, **kwargs)`
+  - `study`: Optuna study with a sampler that uses pyribs. This function will plot the result archive from the sampler's scheduler.
+  - ax: Axes on which to plot the heatmap. If None, we retrieve the current axes.
+  - kwargs: All remaining kwargs will be passed to [`grid_archive_heatmap`](https://docs.pyribs.org/en/stable/api/ribs.visualize.grid_archive_heatmap.html).
```
Comment (nabenabe0928) [inline package/visualization/plot_pyribs/README.md:17]:
```suggestion
- `plot_grid_archive_heatmap(study: optuna.Study, ax: plt.Axes, **kwargs)`
  - `study`: Optuna study with a sampler that uses pyribs. This function will plot the result archive from the sampler's scheduler.
  - `ax`: Axes on which to plot the heatmap. If None, we retrieve the current axes.
  - `**kwargs`: All remaining kwargs will be passed to [`grid_archive_heatmap`](https://docs.pyribs.org/en/stable/api/ribs.visualize.grid_archive_heatmap.html).
```

File: package/visualization/plot_pyribs/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+tags: [visualization, quality diversity, pyribs]
+optuna_versions: [4.0.0]
+license: MIT License
+---
+
+## Class or Function Names
+
+- `plot_grid_archive_heatmap()`
+
+- `plot_grid_archive_heatmap(study: optuna.Study, ax: plt.Axes, **kwargs)`
```
Comment (nabenabe0928) [inline package/visualization/plot_pyribs/README.md:14]:
```suggestion
- `plot_grid_archive_heatmap(study: optuna.Study, ax: plt.Axes, **kwargs)`
```

File: package/visualization/plot_pyribs/README.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+description: This visualizaton module provides wrappers around the visualization functions from pyribs, which is useful for plotting results from CmaMaeSampler.
+tags: [visualization, quality diversity, pyribs]
+optuna_versions: [4.0.0]
+license: MIT License
+---
+
+## Class or Function Names
+
+- `plot_grid_archive_heatmap(study: optuna.Study, ax: plt.Axes, **kwargs)`
+  - `study`: Optuna study with a sampler that uses pyribs. This function will plot the result archive from the sampler's scheduler.
```
Comment (nabenabe0928) [inline package/visualization/plot_pyribs/README.md:13]:
```suggestion

  - `study`: Optuna study with a sampler that uses pyribs. This function will plot the result archive from the sampler's scheduler.
```

File: package/visualization/plot_pyribs/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+## Class or Function Names
+
+- `plot_grid_archive_heatmap(study: optuna.Study, ax: plt.Axes, **kwargs)`
+
+  - `study`: Optuna study with a sampler that uses pyribs. This function will plot the result archive from the sampler's scheduler.
+  - `ax`: Axes on which to plot the heatmap. If None, we retrieve the current axes.
+  - `**kwargs`: All remaining kwargs will be passed to [`grid_archive_heatmap`](https://docs.pyribs.org/en/stable/api/ribs.visualize.grid_archive_heatmap.html).
+
+
```
Comment (nabenabe0928) [inline package/visualization/plot_pyribs/README.md:18]:
```suggestion
```

**Pull request merged**